RecoCTPPS/ProtonReconstruction/plugins/CTPPSProtonProducer.cc

```diff
@@ -388,11 +391,12 @@ void CTPPSProtonProducer::produce(edm::Event &iEvent, const edm::EventSetup &iSe
               ssLog << std::endl
                     << "* reconstruction from tracking-RP tracks: " << i << ", " << j << " and timing-RP tracks: ";
 
-            // process tracking-RP data
+            // buffer contributing tracks
             CTPPSLocalTrackLiteRefVector sel_tracks;
             sel_tracks.push_back(CTPPSLocalTrackLiteRef(hTracks, i));
             sel_tracks.push_back(CTPPSLocalTrackLiteRef(hTracks, j));
-            reco::ForwardProton proton = algorithm_.reconstructFromMultiRP(sel_tracks, *hLHCInfo, ssLog);
+
+            CTPPSLocalTrackLiteRefVector sel_track_for_kin_reco = sel_tracks;
```

Comment:
What is the reason for duplicatying (copying) the verctor of tracks and move forward the proton reconstruction?
If it is only for making clear that only a subset of sel_track's is used in the proton reco perhaps one line of comment would have been enough

Reply:
The kinematics reconstruction (reconstructFromMultiRP) shall only use local tracks from tracking RPs (the spatial segmentation of the timing sensors is too coarse). However, the reco-proton object should also "remember" which timing track(s) contributed (by time information, not kinematics). Thus the separation of the two "selections". On practical level, the copy on line 399 copies two references, thus should not be costly.

Reply:
Alternatively, one could move the invocation of reconstructFromMultiRP to line ~399. This would avoid copying the selection vector, however it breaks the debug print out of contributing tracks (lines 392 and 411).

Reply:
> Alternatively, one could move the invocation of reconstructFromMultiRP to line ~399. This would avoid copying the selection vector, however it breaks the debug print out of contributing tracks (lines 392 and 411).

Ok, this is how it was before your update. Therefore, the answer to my question is "because keeping it as it was before it would broke the debug print out of contributing tracks", which makes sense.